test/e2e/user_defined_sa_test.go

```diff
@@ -0,0 +1,334 @@
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+)
+
+func TestUserDefinedServiceAccountWithNoPermission(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, false)
+
+	kubeclient := newKubeClient(t)
+	crclient := newCRClient(t)
+
+	namespace := genName("scoped-ns-")
+	_, ceanupNS := newNamespace(t, kubeclient, namespace)
+	defer ceanupNS()
+
+	// Create a service account, but add no permission to it.
+	saName := genName("scoped-sa-")
+	_, cleanupSA := newServiceAccount(t, kubeclient, namespace, saName)
+	defer cleanupSA()
+
+	// Add an OperatorGroup and specify the service account.
+	ogName := genName("scoped-og-")
+	_, cleanupOG := newOperatorGroupWithServiceAccount(t, crclient, namespace, ogName, saName)
+	defer cleanupOG()
+
+	catsrc, subSpec, catsrcCleanup := newCatalogSource(t, kubeclient, crclient, namespace)
+	defer catsrcCleanup()
+
+	// Ensure that the catalog source is resolved before we create a subscription.
+	_, err := fetchCatalogSource(t, crclient, catsrc.GetName(), namespace, catalogSourceRegistryPodSynced)
+	require.NoError(t, err)
+
+	subscriptionName := genName("scoped-sub-")
+	cleanupSubscription := createSubscriptionForCatalog(t, crclient, namespace, subscriptionName, catsrc.GetName(), subSpec.Package, subSpec.Channel, subSpec.StartingCSV, subSpec.InstallPlanApproval)
+	defer cleanupSubscription()
+
+	// Wait until an install plan is created.
+	subscription, err := fetchSubscription(t, crclient, namespace, subscriptionName, subscriptionHasInstallPlanChecker)
+	require.NoError(t, err)
+	require.NotNil(t, subscription)
+
+	// We expect the InstallPlan to be in status: Failed.
+	ipName := subscription.Status.Install.Name
+	ipPhaseCheckerFunc := buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseFailed)
+	ipGot, err := fetchInstallPlanWithNamespace(t, crclient, ipName, namespace, ipPhaseCheckerFunc)
+	require.NoError(t, err)
+
+	conditionGot := mustHaveCondition(t, ipGot, v1alpha1.InstallPlanInstalled)
+	assert.Equal(t, corev1.ConditionFalse, conditionGot.Status)
+	assert.Equal(t, v1alpha1.InstallPlanReasonComponentFailed, conditionGot.Reason)
+	assert.Contains(t, conditionGot.Message, fmt.Sprintf("is forbidden: User \"system:serviceaccount:%s:%s\" cannot create resource", namespace, saName))
+
+	// Verify that all step resources are in Unknown state.
+	for _, step := range ipGot.Status.Plan {
```

Comment:
Do we give the user enough information to understand that this is a permission issue?

Reply:
The install plan status says `is forbidden`, it specifies the `actor`, `action` and `resource` I think. Here is an assert I have on e2e test. 

```
assert.Contains(t, conditionGot.Message, fmt.Sprintf("is forbidden: User \"system:serviceaccount:%s:%s\" cannot create resource", namespace, saName))
```

Reply:
This can be a follow up, but it's probably worth mentioning where we got that serviceaccount from (the operatorgroup object)

Reply:
I have added this to the jira ticket so we can track it.